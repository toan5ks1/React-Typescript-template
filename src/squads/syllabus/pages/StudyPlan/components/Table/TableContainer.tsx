import { styled } from "@mui/material/styles";

const PREFIX = "StudyPlanItemTable";

export const classes = {
    loNameColumn: `${PREFIX}-loNameColumn`,
    dateColumn: `${PREFIX}-dateColumn`,
    topicColumn: `${PREFIX}-topicColumn`,
    dateEditColumn: `${PREFIX}-dateEditColumn`,
    topicEditColumn: `${PREFIX}-topicEditColumn`,
    visibilityEditColumn: `${PREFIX}-visibilityEditColumn`,
};

export const TableContainer = styled("div")(({ theme }) => ({
    [`& .${classes.loNameColumn}`]: {
        paddingTop: theme.spacing(1.25),
        paddingBottom: theme.spacing(1.25),
    },

    [`& .${classes.dateColumn}`]: {
        width: "13.8%",
    },

    [`& .${classes.topicColumn}`]: {
        width: "17.2%",
    },

    [`& .${classes.dateEditColumn}`]: {
        width: "15.8%",
    },

    [`& .${classes.topicEditColumn}`]: {
        width: "13%",
    },

    [`& .${classes.visibilityEditColumn}`]: {
        width: "1%",
        whiteSpace: "nowrap",
        textAlign: "center",
    },
}));

TableContainer.displayName = "TableContainer";
