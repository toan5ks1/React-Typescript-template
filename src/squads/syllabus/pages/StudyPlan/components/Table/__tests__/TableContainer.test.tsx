import { TableContainer } from "../TableContainer";

import { render } from "@testing-library/react";

describe(TableContainer.name, () => {
    it("should render correctly", () => {
        const { baseElement } = render(<TableContainer />);
        expect(baseElement).toBeTruthy();
        expect(baseElement).toMatchSnapshot();
    });
});
