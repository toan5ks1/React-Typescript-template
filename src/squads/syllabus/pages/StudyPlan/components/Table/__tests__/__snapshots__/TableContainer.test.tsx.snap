// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-0 .StudyPlanItemTable-loNameColumn {
  padding-top: 10px;
  padding-bottom: 10px;
}

.emotion-0 .StudyPlanItemTable-dateColumn {
  width: 13.8%;
}

.emotion-0 .StudyPlanItemTable-topicColumn {
  width: 17.2%;
}

.emotion-0 .StudyPlanItemTable-dateEditColumn {
  width: 15.8%;
}

.emotion-0 .StudyPlanItemTable-topicEditColumn {
  width: 13%;
}

.emotion-0 .StudyPlanItemTable-visibilityEditColumn {
  width: 1%;
  white-space: nowrap;
  text-align: center;
}

<body>
  <div>
    <div
      class="emotion-0"
    />
  </div>
</body>
`;
